horreum-api/src/main/java/io/hyperfoil/tools/horreum/api/data/datastore/CollectorApiDatastoreConfig.java

```diff
@@ -3,26 +3,30 @@
 import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
 import org.eclipse.microprofile.openapi.annotations.media.Schema;
 
+import io.hyperfoil.tools.horreum.api.data.datastore.auth.APIKeyAuth;
+import io.hyperfoil.tools.horreum.api.data.datastore.auth.NoAuth;
+
 @Schema(type = SchemaType.OBJECT, required = true, description = "Type of backend datastore")
 public class CollectorApiDatastoreConfig extends BaseDatastoreConfig {
 
+    public static final String[] auths = { NoAuth._TYPE, APIKeyAuth._TYPE };
+    public static final String name = "Collectorapi";
+    public static final String label = "Collector API";
+    public static final Boolean builtIn = false;
+
     public CollectorApiDatastoreConfig() {
-        super(false);
     }
 
-    @Schema(type = SchemaType.STRING, required = true, description = "Collector API KEY")
-    public String apiKey;
-
     @Schema(type = SchemaType.STRING, required = true, description = "Collector url, e.g. https://collector.foci.life/api/v1/image-stats")
     public String url;
 
     @Override
     public String validateConfig() {
-        if ("".equals(apiKey)) {
-            return "apiKey must be set";
-        }
-        if ("".equals(url)) {
-            return "url must be set";
+        if (authentication instanceof APIKeyAuth) {
+            APIKeyAuth apiKeyAuth = (APIKeyAuth) authentication;
+            if ("".equals(apiKeyAuth.apiKey) || apiKeyAuth.apiKey == null) {
```

Comment:
```suggestion
            if (apiKeyAuth.apiKey == null || apiKeyAuth.apiKey.isEmpty()) {
```

Reply:
will change to `isBlank()` to take into account white-space as well